session/session.go

```diff
@@ -2061,16 +2061,24 @@ func resetCTEStorageMap(se *session) error {
 		return errors.New("type assertion for CTEStorageMap failed")
 	}
 	for _, v := range storageMap {
-		// No need to lock IterInTbl.
-		v.ResTbl.Lock()
-		defer v.ResTbl.Unlock()
-		err1 := v.ResTbl.DerefAndClose()
-		err2 := v.IterInTbl.DerefAndClose()
-		if err1 != nil {
-			return err1
-		}
-		if err2 != nil {
-			return err2
+		// Wrap the loop body in an anonymous function to make sure
+		// the deferred unlock is triggered on every loop iteration.
+		err := func() error {
```

Comment:
It would be better to 
```
v.ResTbl.Lock()
err1 := v.ResTbl.DerefAndClose()
v.ResTbl.Unlock()
```
instead of using a function and defer.

Reply:
Good point. As long as the loop body stays simple, we can, indeed, unlock right away.
Made a change.